[All] Run clang-tidy and update license headers

guparan opened the pull request:
**WARNING: This PR will surely conflict with your dev branches**

Commands executed:
`run-clang-tidy.py -header-filter='.*' -checks='-*,modernize-use-override' -fix`
`run-clang-tidy.py -header-filter='.*' -checks='-*,modernize-use-nullptr' -fix`
`licenseUpdater.sh . auto 2019 "development version"`

With the help of this blog article: https://www.kdab.com/clang-tidy-part-1-modernize-source-code-using-c11c14/

______________________________________________________
<!--- Please leave this at the end of your message -->
This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

damienmarchal commented:
Yes. Thanks

guparan commented:
Now is the good time to suggest any other clang-tidy run among:

| _ | _ |
|------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| [boost-use-to-string](https://clang.llvm.org/extra/clang-tidy/checks/boost-use-to-string.html) | [hicpp-braces-around-statements](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-braces-around-statements.html) |
| [bugprone-argument-comment](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-argument-comment.html) | [hicpp-deprecated-headers](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-deprecated-headers.html) |
| [bugprone-assert-side-effect](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-assert-side-effect.html) | [hicpp-exception-baseclass](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-exception-baseclass.html) |
| [bugprone-bool-pointer-implicit-conversion](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-bool-pointer-implicit-conversion.html) | [hicpp-explicit-conversions](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-explicit-conversions.html) |
| [bugprone-copy-constructor-init](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-copy-constructor-init.html) | [hicpp-function-size](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-function-size.html) |
| [bugprone-dangling-handle](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-dangling-handle.html) | [hicpp-invalid-access-moved](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-invalid-access-moved.html) |
| [bugprone-fold-init-type](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-fold-init-type.html) | [hicpp-member-init](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-member-init.html) |
| [bugprone-forward-declaration-namespace](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-forward-declaration-namespace.html) | [hicpp-move-const-arg](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-move-const-arg.html) |
| [bugprone-inaccurate-erase](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-inaccurate-erase.html) | [hicpp-named-parameter](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-named-parameter.html) |
| [bugprone-integer-division](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-integer-division.html) | [hicpp-new-delete-operators](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-new-delete-operators.html) |
| [bugprone-misplaced-operator-in-strlen-in-alloc](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-misplaced-operator-in-strlen-in-alloc.html) | [hicpp-no-array-decay](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-no-array-decay.html) |
| [bugprone-move-forwarding-reference](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-move-forwarding-reference.html) | [hicpp-no-assembler](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-no-assembler.html) |
| [bugprone-multiple-statement-macro](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-multiple-statement-macro.html) | [hicpp-no-malloc](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-no-malloc.html) |
| [bugprone-string-constructor](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-string-constructor.html) | [hicpp-noexcept-move](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-noexcept-move.html) |
| [bugprone-suspicious-memset-usage](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-suspicious-memset-usage.html) | [hicpp-signed-bitwise](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-signed-bitwise.html) |
| [bugprone-undefined-memory-manipulation](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-undefined-memory-manipulation.html) | [hicpp-special-member-functions](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-special-member-functions.html) |
| [bugprone-use-after-move](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-use-after-move.html) | [hicpp-static-assert](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-static-assert.html) |
| [bugprone-virtual-near-miss](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-virtual-near-miss.html) | [hicpp-undelegated-constructor](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-undelegated-constructor.html) |
| [cert-dcl03-c](https://clang.llvm.org/extra/clang-tidy/checks/cert-dcl03-c.html) | [hicpp-use-auto](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-use-auto.html) |
| [cert-dcl21-cpp](https://clang.llvm.org/extra/clang-tidy/checks/cert-dcl21-cpp.html) | [hicpp-use-emplace](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-use-emplace.html) |
| [cert-dcl50-cpp](https://clang.llvm.org/extra/clang-tidy/checks/cert-dcl50-cpp.html) | [hicpp-use-equals-default](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-use-equals-default.html) |
| [cert-dcl54-cpp](https://clang.llvm.org/extra/clang-tidy/checks/cert-dcl54-cpp.html) | [hicpp-use-equals-delete](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-use-equals-delete.html) |
| [cert-dcl58-cpp](https://clang.llvm.org/extra/clang-tidy/checks/cert-dcl58-cpp.html) | [hicpp-use-noexcept](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-use-noexcept.html) |
| [cert-dcl59-cpp](https://clang.llvm.org/extra/clang-tidy/checks/cert-dcl59-cpp.html) | [hicpp-use-nullptr](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-use-nullptr.html) |
| [cert-env33-c](https://clang.llvm.org/extra/clang-tidy/checks/cert-env33-c.html) | [hicpp-use-override](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-use-override.html) |
| [cert-err09-cpp](https://clang.llvm.org/extra/clang-tidy/checks/cert-err09-cpp.html) | [hicpp-vararg](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-vararg.html) |
| [cert-err34-c](https://clang.llvm.org/extra/clang-tidy/checks/cert-err34-c.html) | [llvm-header-guard](https://clang.llvm.org/extra/clang-tidy/checks/llvm-header-guard.html) |
| [cert-err52-cpp](https://clang.llvm.org/extra/clang-tidy/checks/cert-err52-cpp.html) | [llvm-include-order](https://clang.llvm.org/extra/clang-tidy/checks/llvm-include-order.html) |
| [cert-err58-cpp](https://clang.llvm.org/extra/clang-tidy/checks/cert-err58-cpp.html) | [llvm-namespace-comment](https://clang.llvm.org/extra/clang-tidy/checks/llvm-namespace-comment.html) |
| [cert-err60-cpp](https://clang.llvm.org/extra/clang-tidy/checks/cert-err60-cpp.html) | [llvm-twine-local](https://clang.llvm.org/extra/clang-tidy/checks/llvm-twine-local.html) |
| [cert-err61-cpp](https://clang.llvm.org/extra/clang-tidy/checks/cert-err61-cpp.html) | [misc-definitions-in-headers](https://clang.llvm.org/extra/clang-tidy/checks/misc-definitions-in-headers.html) |
| [cert-fio38-c](https://clang.llvm.org/extra/clang-tidy/checks/cert-fio38-c.html) | [misc-forwarding-reference-overload](https://clang.llvm.org/extra/clang-tidy/checks/misc-forwarding-reference-overload.html) |
| [cert-flp30-c](https://clang.llvm.org/extra/clang-tidy/checks/cert-flp30-c.html) | [misc-incorrect-roundings](https://clang.llvm.org/extra/clang-tidy/checks/misc-incorrect-roundings.html) |
| [cert-msc30-c](https://clang.llvm.org/extra/clang-tidy/checks/cert-msc30-c.html) | [misc-lambda-function-name](https://clang.llvm.org/extra/clang-tidy/checks/misc-lambda-function-name.html) |
| [cert-msc50-cpp](https://clang.llvm.org/extra/clang-tidy/checks/cert-msc50-cpp.html) | [misc-macro-parentheses](https://clang.llvm.org/extra/clang-tidy/checks/misc-macro-parentheses.html) |
| [cert-oop11-cpp](https://clang.llvm.org/extra/clang-tidy/checks/cert-oop11-cpp.html) | [misc-macro-repeated-side-effects](https://clang.llvm.org/extra/clang-tidy/checks/misc-macro-repeated-side-effects.html) |
| [clang-analyzer-apiModeling.google.GTest](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-apiModeling.google.GTest.html) | [misc-misplaced-const](https://clang.llvm.org/extra/clang-tidy/checks/misc-misplaced-const.html) |
| [clang-analyzer-core.CallAndMessage](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.CallAndMessage.html) | [misc-misplaced-widening-cast](https://clang.llvm.org/extra/clang-tidy/checks/misc-misplaced-widening-cast.html) |
| [clang-analyzer-core.DivideZero](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.DivideZero.html) | [misc-new-delete-overloads](https://clang.llvm.org/extra/clang-tidy/checks/misc-new-delete-overloads.html) |
| [clang-analyzer-core.DynamicTypePropagation](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.DynamicTypePropagation.html) | [misc-non-copyable-objects](https://clang.llvm.org/extra/clang-tidy/checks/misc-non-copyable-objects.html) |
| [clang-analyzer-core.NonNullParamChecker](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.NonNullParamChecker.html) | [misc-redundant-expression](https://clang.llvm.org/extra/clang-tidy/checks/misc-redundant-expression.html) |
| [clang-analyzer-core.NonnilStringConstants](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.NonnilStringConstants.html) | [misc-sizeof-container](https://clang.llvm.org/extra/clang-tidy/checks/misc-sizeof-container.html) |
| [clang-analyzer-core.NullDereference](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.NullDereference.html) | [misc-sizeof-expression](https://clang.llvm.org/extra/clang-tidy/checks/misc-sizeof-expression.html) |
| [clang-analyzer-core.StackAddressEscape](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.StackAddressEscape.html) | [misc-static-assert](https://clang.llvm.org/extra/clang-tidy/checks/misc-static-assert.html) |
| [clang-analyzer-core.UndefinedBinaryOperatorResult](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.UndefinedBinaryOperatorResult.html) | [misc-string-compare](https://clang.llvm.org/extra/clang-tidy/checks/misc-string-compare.html) |
| [clang-analyzer-core.VLASize](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.VLASize.html) | [misc-string-integer-assignment](https://clang.llvm.org/extra/clang-tidy/checks/misc-string-integer-assignment.html) |
| [clang-analyzer-core.builtin.BuiltinFunctions](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.builtin.BuiltinFunctions.html) | [misc-string-literal-with-embedded-nul](https://clang.llvm.org/extra/clang-tidy/checks/misc-string-literal-with-embedded-nul.html) |
| [clang-analyzer-core.builtin.NoReturnFunctions](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.builtin.NoReturnFunctions.html) | [misc-suspicious-enum-usage](https://clang.llvm.org/extra/clang-tidy/checks/misc-suspicious-enum-usage.html) |
| [clang-analyzer-core.uninitialized.ArraySubscript](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.uninitialized.ArraySubscript.html) | [misc-suspicious-missing-comma](https://clang.llvm.org/extra/clang-tidy/checks/misc-suspicious-missing-comma.html) |
| [clang-analyzer-core.uninitialized.Assign](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.uninitialized.Assign.html) | [misc-suspicious-semicolon](https://clang.llvm.org/extra/clang-tidy/checks/misc-suspicious-semicolon.html) |
| [clang-analyzer-core.uninitialized.Branch](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.uninitialized.Branch.html) | [misc-suspicious-string-compare](https://clang.llvm.org/extra/clang-tidy/checks/misc-suspicious-string-compare.html) |
| [clang-analyzer-core.uninitialized.CapturedBlockVariable](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.uninitialized.CapturedBlockVariable.html) | [misc-swapped-arguments](https://clang.llvm.org/extra/clang-tidy/checks/misc-swapped-arguments.html) |
| [clang-analyzer-core.uninitialized.UndefReturn](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-core.uninitialized.UndefReturn.html) | [misc-throw-by-value-catch-by-reference](https://clang.llvm.org/extra/clang-tidy/checks/misc-throw-by-value-catch-by-reference.html) |
| [clang-analyzer-cplusplus.NewDelete](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-cplusplus.NewDelete.html) | [misc-unconventional-assign-operator](https://clang.llvm.org/extra/clang-tidy/checks/misc-unconventional-assign-operator.html) |
| [clang-analyzer-cplusplus.NewDeleteLeaks](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-cplusplus.NewDeleteLeaks.html) | [misc-undelegated-constructor](https://clang.llvm.org/extra/clang-tidy/checks/misc-undelegated-constructor.html) |
| [clang-analyzer-cplusplus.SelfAssignment](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-cplusplus.SelfAssignment.html) | [misc-uniqueptr-reset-release](https://clang.llvm.org/extra/clang-tidy/checks/misc-uniqueptr-reset-release.html) |
| [clang-analyzer-deadcode.DeadStores](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-deadcode.DeadStores.html) | [misc-unused-alias-decls](https://clang.llvm.org/extra/clang-tidy/checks/misc-unused-alias-decls.html) |
| [clang-analyzer-llvm.Conventions](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-llvm.Conventions.html) | [misc-unused-parameters](https://clang.llvm.org/extra/clang-tidy/checks/misc-unused-parameters.html) |
| [clang-analyzer-nullability.NullPassedToNonnull](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-nullability.NullPassedToNonnull.html) | [misc-unused-raii](https://clang.llvm.org/extra/clang-tidy/checks/misc-unused-raii.html) |
| [clang-analyzer-nullability.NullReturnedFromNonnull](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-nullability.NullReturnedFromNonnull.html) | [misc-unused-using-decls](https://clang.llvm.org/extra/clang-tidy/checks/misc-unused-using-decls.html) |
| [clang-analyzer-nullability.NullableDereferenced](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-nullability.NullableDereferenced.html) | [modernize-avoid-bind](https://clang.llvm.org/extra/clang-tidy/checks/modernize-avoid-bind.html) |
| [clang-analyzer-nullability.NullablePassedToNonnull](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-nullability.NullablePassedToNonnull.html) | [modernize-deprecated-headers](https://clang.llvm.org/extra/clang-tidy/checks/modernize-deprecated-headers.html) |
| [clang-analyzer-nullability.NullableReturnedFromNonnull](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-nullability.NullableReturnedFromNonnull.html) | [modernize-loop-convert](https://clang.llvm.org/extra/clang-tidy/checks/modernize-loop-convert.html) |
| [clang-analyzer-optin.cplusplus.VirtualCall](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-optin.cplusplus.VirtualCall.html) | [modernize-make-shared](https://clang.llvm.org/extra/clang-tidy/checks/modernize-make-shared.html) |
| [clang-analyzer-optin.mpi.MPI-Checker](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-optin.mpi.MPI-Checker.html) | [modernize-make-unique](https://clang.llvm.org/extra/clang-tidy/checks/modernize-make-unique.html) |
| [clang-analyzer-optin.performance.Padding](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-optin.performance.Padding.html) | [modernize-pass-by-value](https://clang.llvm.org/extra/clang-tidy/checks/modernize-pass-by-value.html) |
| [clang-analyzer-optin.portability.UnixAPI](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-optin.portability.UnixAPI.html) | [modernize-raw-string-literal](https://clang.llvm.org/extra/clang-tidy/checks/modernize-raw-string-literal.html) |
| [clang-analyzer-security.FloatLoopCounter](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-security.FloatLoopCounter.html) | [modernize-redundant-void-arg](https://clang.llvm.org/extra/clang-tidy/checks/modernize-redundant-void-arg.html) |
| [clang-analyzer-security.insecureAPI.UncheckedReturn](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-security.insecureAPI.UncheckedReturn.html) | [modernize-replace-auto-ptr](https://clang.llvm.org/extra/clang-tidy/checks/modernize-replace-auto-ptr.html) |
| [clang-analyzer-security.insecureAPI.getpw](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-security.insecureAPI.getpw.html) | [modernize-replace-random-shuffle](https://clang.llvm.org/extra/clang-tidy/checks/modernize-replace-random-shuffle.html) |
| [clang-analyzer-security.insecureAPI.gets](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-security.insecureAPI.gets.html) | [modernize-return-braced-init-list](https://clang.llvm.org/extra/clang-tidy/checks/modernize-return-braced-init-list.html) |
| [clang-analyzer-security.insecureAPI.mkstemp](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-security.insecureAPI.mkstemp.html) | [modernize-shrink-to-fit](https://clang.llvm.org/extra/clang-tidy/checks/modernize-shrink-to-fit.html) |
| [clang-analyzer-security.insecureAPI.mktemp](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-security.insecureAPI.mktemp.html) | [modernize-unary-static-assert](https://clang.llvm.org/extra/clang-tidy/checks/modernize-unary-static-assert.html) |
| [clang-analyzer-security.insecureAPI.rand](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-security.insecureAPI.rand.html) | [modernize-use-auto](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-auto.html) |
| [clang-analyzer-security.insecureAPI.strcpy](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-security.insecureAPI.strcpy.html) | [modernize-use-bool-literals](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-bool-literals.html) |
| [clang-analyzer-security.insecureAPI.vfork](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-security.insecureAPI.vfork.html) | [modernize-use-default-member-init](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-default-member-init.html) |
| [clang-analyzer-unix.API](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-unix.API.html) | [modernize-use-emplace](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-emplace.html) |
| [clang-analyzer-unix.Malloc](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-unix.Malloc.html) | [modernize-use-equals-default](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-equals-default.html) |
| [clang-analyzer-unix.MallocSizeof](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-unix.MallocSizeof.html) | [modernize-use-equals-delete](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-equals-delete.html) |
| [clang-analyzer-unix.MismatchedDeallocator](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-unix.MismatchedDeallocator.html) | [modernize-use-noexcept](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-noexcept.html) |
| [clang-analyzer-unix.StdCLibraryFunctions](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-unix.StdCLibraryFunctions.html) | [modernize-use-nullptr](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-nullptr.html) |
| [clang-analyzer-unix.Vfork](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-unix.Vfork.html) | [modernize-use-override](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-override.html) |
| [clang-analyzer-unix.cstring.BadSizeArg](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-unix.cstring.BadSizeArg.html) | [modernize-use-transparent-functors](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-transparent-functors.html) |
| [clang-analyzer-unix.cstring.NullArg](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-unix.cstring.NullArg.html) | [modernize-use-using](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-using.html) |
| [clang-analyzer-valist.CopyToSelf](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-valist.CopyToSelf.html) | [mpi-buffer-deref](https://clang.llvm.org/extra/clang-tidy/checks/mpi-buffer-deref.html) |
| [clang-analyzer-valist.Uninitialized](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-valist.Uninitialized.html) | [mpi-type-mismatch](https://clang.llvm.org/extra/clang-tidy/checks/mpi-type-mismatch.html) |
| [clang-analyzer-valist.Unterminated](https://clang.llvm.org/extra/clang-tidy/checks/clang-analyzer-valist.Unterminated.html) | [objc-avoid-nserror-init](https://clang.llvm.org/extra/clang-tidy/checks/objc-avoid-nserror-init.html) |
| [cppcoreguidelines-c-copy-assignment-signature](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-c-copy-assignment-signature.html) | [objc-avoid-spinlock](https://clang.llvm.org/extra/clang-tidy/checks/objc-avoid-spinlock.html) |
| [cppcoreguidelines-interfaces-global-init](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-interfaces-global-init.html) | [objc-forbidden-subclassing](https://clang.llvm.org/extra/clang-tidy/checks/objc-forbidden-subclassing.html) |
| [cppcoreguidelines-no-malloc](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-no-malloc.html) | [objc-property-declaration](https://clang.llvm.org/extra/clang-tidy/checks/objc-property-declaration.html) |
| [cppcoreguidelines-owning-memory](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-owning-memory.html) | [performance-faster-string-find](https://clang.llvm.org/extra/clang-tidy/checks/performance-faster-string-find.html) |
| [cppcoreguidelines-pro-bounds-array-to-pointer-decay](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-bounds-array-to-pointer-decay.html) | [performance-for-range-copy](https://clang.llvm.org/extra/clang-tidy/checks/performance-for-range-copy.html) |
| [cppcoreguidelines-pro-bounds-constant-array-index](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-bounds-constant-array-index.html) | [performance-implicit-conversion-in-loop](https://clang.llvm.org/extra/clang-tidy/checks/performance-implicit-conversion-in-loop.html) |
| [cppcoreguidelines-pro-bounds-pointer-arithmetic](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-bounds-pointer-arithmetic.html) | [performance-inefficient-algorithm](https://clang.llvm.org/extra/clang-tidy/checks/performance-inefficient-algorithm.html) |
| [cppcoreguidelines-pro-type-const-cast](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-type-const-cast.html) | [performance-inefficient-string-concatenation](https://clang.llvm.org/extra/clang-tidy/checks/performance-inefficient-string-concatenation.html) |
| [cppcoreguidelines-pro-type-cstyle-cast](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-type-cstyle-cast.html) | [performance-inefficient-vector-operation](https://clang.llvm.org/extra/clang-tidy/checks/performance-inefficient-vector-operation.html) |
| [cppcoreguidelines-pro-type-member-init](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-type-member-init.html) | [performance-move-const-arg](https://clang.llvm.org/extra/clang-tidy/checks/performance-move-const-arg.html) |
| [cppcoreguidelines-pro-type-reinterpret-cast](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-type-reinterpret-cast.html) | [performance-move-constructor-init](https://clang.llvm.org/extra/clang-tidy/checks/performance-move-constructor-init.html) |
| [cppcoreguidelines-pro-type-static-cast-downcast](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-type-static-cast-downcast.html) | [performance-noexcept-move-constructor](https://clang.llvm.org/extra/clang-tidy/checks/performance-noexcept-move-constructor.html) |
| [cppcoreguidelines-pro-type-union-access](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-type-union-access.html) | [performance-type-promotion-in-math-fn](https://clang.llvm.org/extra/clang-tidy/checks/performance-type-promotion-in-math-fn.html) |
| [cppcoreguidelines-pro-type-vararg](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-type-vararg.html) | [performance-unnecessary-copy-initialization](https://clang.llvm.org/extra/clang-tidy/checks/performance-unnecessary-copy-initialization.html) |
| [cppcoreguidelines-slicing](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-slicing.html) | [performance-unnecessary-value-param](https://clang.llvm.org/extra/clang-tidy/checks/performance-unnecessary-value-param.html) |
| [cppcoreguidelines-special-member-functions](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-special-member-functions.html) | [readability-avoid-const-params-in-decls](https://clang.llvm.org/extra/clang-tidy/checks/readability-avoid-const-params-in-decls.html) |
| [fuchsia-default-arguments](https://clang.llvm.org/extra/clang-tidy/checks/fuchsia-default-arguments.html) | [readability-braces-around-statements](https://clang.llvm.org/extra/clang-tidy/checks/readability-braces-around-statements.html) |
| [fuchsia-overloaded-operator](https://clang.llvm.org/extra/clang-tidy/checks/fuchsia-overloaded-operator.html) | [readability-container-size-empty](https://clang.llvm.org/extra/clang-tidy/checks/readability-container-size-empty.html) |
| [fuchsia-virtual-inheritance](https://clang.llvm.org/extra/clang-tidy/checks/fuchsia-virtual-inheritance.html) | [readability-delete-null-pointer](https://clang.llvm.org/extra/clang-tidy/checks/readability-delete-null-pointer.html) |
| [google-build-explicit-make-pair](https://clang.llvm.org/extra/clang-tidy/checks/google-build-explicit-make-pair.html) | [readability-deleted-default](https://clang.llvm.org/extra/clang-tidy/checks/readability-deleted-default.html) |
| [google-build-namespaces](https://clang.llvm.org/extra/clang-tidy/checks/google-build-namespaces.html) | [readability-else-after-return](https://clang.llvm.org/extra/clang-tidy/checks/readability-else-after-return.html) |
| [google-build-using-namespace](https://clang.llvm.org/extra/clang-tidy/checks/google-build-using-namespace.html) | [readability-function-size](https://clang.llvm.org/extra/clang-tidy/checks/readability-function-size.html) |
| [google-default-arguments](https://clang.llvm.org/extra/clang-tidy/checks/google-default-arguments.html) | [readability-identifier-naming](https://clang.llvm.org/extra/clang-tidy/checks/readability-identifier-naming.html) |
| [google-explicit-constructor](https://clang.llvm.org/extra/clang-tidy/checks/google-explicit-constructor.html) | [readability-implicit-bool-conversion](https://clang.llvm.org/extra/clang-tidy/checks/readability-implicit-bool-conversion.html) |
| [google-global-names-in-headers](https://clang.llvm.org/extra/clang-tidy/checks/google-global-names-in-headers.html) | [readability-inconsistent-declaration-parameter-name](https://clang.llvm.org/extra/clang-tidy/checks/readability-inconsistent-declaration-parameter-name.html) |
| [google-objc-avoid-throwing-exception](https://clang.llvm.org/extra/clang-tidy/checks/google-objc-avoid-throwing-exception.html) | [readability-misleading-indentation](https://clang.llvm.org/extra/clang-tidy/checks/readability-misleading-indentation.html) |
| [google-objc-global-variable-declaration](https://clang.llvm.org/extra/clang-tidy/checks/google-objc-global-variable-declaration.html) | [readability-misplaced-array-index](https://clang.llvm.org/extra/clang-tidy/checks/readability-misplaced-array-index.html) |
| [google-readability-braces-around-statements](https://clang.llvm.org/extra/clang-tidy/checks/google-readability-braces-around-statements.html) | [readability-named-parameter](https://clang.llvm.org/extra/clang-tidy/checks/readability-named-parameter.html) |
| [google-readability-casting](https://clang.llvm.org/extra/clang-tidy/checks/google-readability-casting.html) | [readability-non-const-parameter](https://clang.llvm.org/extra/clang-tidy/checks/readability-non-const-parameter.html) |
| [google-readability-function-size](https://clang.llvm.org/extra/clang-tidy/checks/google-readability-function-size.html) | [readability-redundant-control-flow](https://clang.llvm.org/extra/clang-tidy/checks/readability-redundant-control-flow.html) |
| [google-readability-namespace-comments](https://clang.llvm.org/extra/clang-tidy/checks/google-readability-namespace-comments.html) | [readability-redundant-declaration](https://clang.llvm.org/extra/clang-tidy/checks/readability-redundant-declaration.html) |
| [google-readability-redundant-smartptr-get](https://clang.llvm.org/extra/clang-tidy/checks/google-readability-redundant-smartptr-get.html) | [readability-redundant-function-ptr-dereference](https://clang.llvm.org/extra/clang-tidy/checks/readability-redundant-function-ptr-dereference.html) |
| [google-readability-todo](https://clang.llvm.org/extra/clang-tidy/checks/google-readability-todo.html) | [readability-redundant-member-init](https://clang.llvm.org/extra/clang-tidy/checks/readability-redundant-member-init.html) |
| [google-runtime-int](https://clang.llvm.org/extra/clang-tidy/checks/google-runtime-int.html) | [readability-redundant-smartptr-get](https://clang.llvm.org/extra/clang-tidy/checks/readability-redundant-smartptr-get.html) |
| [google-runtime-member-string-references](https://clang.llvm.org/extra/clang-tidy/checks/google-runtime-member-string-references.html) | [readability-redundant-string-cstr](https://clang.llvm.org/extra/clang-tidy/checks/readability-redundant-string-cstr.html) |
| [google-runtime-operator](https://clang.llvm.org/extra/clang-tidy/checks/google-runtime-operator.html) | [readability-redundant-string-init](https://clang.llvm.org/extra/clang-tidy/checks/readability-redundant-string-init.html) |
| [google-runtime-references](https://clang.llvm.org/extra/clang-tidy/checks/google-runtime-references.html) | [readability-simplify-boolean-expr](https://clang.llvm.org/extra/clang-tidy/checks/readability-simplify-boolean-expr.html) |
|  | [readability-static-accessed-through-instance](https://clang.llvm.org/extra/clang-tidy/checks/readability-static-accessed-through-instance.html) |
|  | [readability-static-definition-in-anonymous-namespace](https://clang.llvm.org/extra/clang-tidy/checks/readability-static-definition-in-anonymous-namespace.html) |
|  | [readability-uniqueptr-delete-release](https://clang.llvm.org/extra/clang-tidy/checks/readability-uniqueptr-delete-release.html) |

hugtalbot committed:
Fix compilation: duplication of constructor/destructor

epernod commented on SofaAdvanced/config.h.in:
```diff
@@ -1,6 +1,6 @@
 /******************************************************************************
 *       SOFA, Simulation Open-Framework Architecture, development version     *
-*                (c) 2006-2018 INRIA, USTL, UJF, CNRS, MGH                    *
+*                (c) 2006-2019 INRIA, USTL, UJF, CNRS, MGH                    *
```
it is not possible to put more than 2019 to avoid changing that each year?

epernod commented on SofaKernel/extlibs/newmat/newmat/newmat.h:
```diff
@@ -361,7 +361,7 @@ class BaseMatrix : public Janitor               // base of all matrix classes
    Real NormInfinity() const;
    virtual MatrixBandWidth BandWidth() const;  // bandwidths of band matrix
-   virtual void CleanUp() {}                   // to clear store
+   void CleanUp() override {}                   // to clear store
```
sure about removing the virtual? I don't think someone inherite from newmat anyway... or I don understand exactly what override means

jnbrunet replied:
override implies virtual.

Normally you have only the base class method with the **virtual** specifier, and all derived classes  have the **override** specifier. It's the **final** specifier that ends the "overriddence" of the method

damienmarchal commented:
To me this PR is to merge once updated.

guparan committed:
[All] Update license headers 2019

guparan committed:
[All] Run clang-tidy for overrides

run-clang-tidy.py -header-filter='.*' -checks='-*,modernize-use-override' -fix

guparan committed:
[All] Run clang-tidy for nullptr

run-clang-tidy.py -header-filter='.*' -checks='-*,modernize-use-nullptr' -fix

guparan committed:
Merge remote-tracking branch 'sofa-framework/master' into clang_tidy_and_license_update

guparan commented:
modernize-deprecated-headers seems to cause problems so I will just revert it for now. We need to merge the rest asap.

EDIT: the problem was not caused by modernize-deprecated-headers so I reapplied it.

guparan committed:
Revert "Fix compilation: duplication of constructor/destructor"

This reverts commit 893e935ff8037da9dd809ac7d0e4bef7f24e048d.

guparan committed:
[All] Run clang-tidy for deprecated-headers

run-clang-tidy.py -header-filter='.*' -checks='-*,modernize-deprecated-headers' -fix

guparan commented:
Replacing `#include <math.h>` by `#include <cmath>` produces a `'M_PI': undeclared identifier` error on Windows for `applications/plugins/sofadistancegrid/components/collision/DistanceGridCollisionModel.h` (and only this one).
Any idea how to solve that?

guparan committed:
[SofaDistanceGrid] FIX M_PI error by rolling back to math.h

guparan committed:
Merge remote-tracking branch 'sofa-framework/master' into clang_tidy_and_license_update

# Conflicts:
#	SofaKernel/framework/sofa/core/visual/DrawToolGL.h

guparan committed:
Merge remote-tracking branch 'sofa-framework/master' into clang_tidy_and_license_update

# Conflicts:
#	applications/plugins/SofaPython/SceneLoaderPY.h
#	modules/SofaGraphComponent/SceneCheckerVisitor.h

guparan commented:
Merged it myself to stop maintaining it :innocent: 
(i'm free now)
